tools/sony9pin/sony9pin.cpp

```diff
@@ -60,30 +60,27 @@ void usage(const string& commandName) {
   }
 }
 
-void print_timecode(const Sony9PinRemote::TimeCode& tc)
+void print_timecode_userbits(bool print_userbits)
 {
+  Sony9PinRemote::TimeCode tc = deck.timecode();
   cerr << "TimeCode: "
        << setw(2) << setfill('0') << (unsigned int)tc.hour << ':'
        << setw(2) << setfill('0') << (unsigned int)tc.minute << ':'
        << setw(2) << setfill('0') << (unsigned int)tc.second << ';'
        << setw(2) << setfill('0') << (unsigned int)tc.frame << ' '
        << "CF: " << (unsigned int)tc.is_cf << ' '
-       << "DF: " << (unsigned int)tc.is_df << '\n';
-}
+       << "DF: " << (unsigned int)tc.is_df;
 
-void print_userbits(const Sony9PinRemote::UserBits& ub)
-{
-  cerr << "UserBits: "
-       << (char)ub.bytes[0] << ' '
-       << (char)ub.bytes[1] << ' '
-       << (char)ub.bytes[2] << ' '
-       << (char)ub.bytes[3] << '\n';
-}
+  if (print_userbits) {
+    Sony9PinRemote::UserBits ub = deck.userbits();
+    cerr << " UB: "
+         << hex << uppercase << setw(2) << setfill('0') << (unsigned int)ub.bytes[0] << ':'
```

Comment:
Could you flip the endianness here. For instance this tape gives:

```
TimeCode: 01:08:14;08 CF: 0 DF: 0 UB: 08:14:08:01
TimeCode: 01:08:14;10 CF: 0 DF: 0 UB: 10:14:08:01
TimeCode: 01:08:14;11 CF: 0 DF: 0 UB: 11:14:08:01
TimeCode: 01:08:14;13 CF: 0 DF: 0 UB: 13:14:08:01
```

When UB should be like `01:08:14:13`

Reply:
Noting at https://github.com/hideakitai/Sony9PinRemote/blob/20227fa426bc66fc182333bfbb0e9156a91b86e5/Sony9PinRemote/Decoder.h#L697-L700 that the timecode values are in little endian order as well.